packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts

```diff
@@ -47,3 +58,77 @@ const commandlet = new SfdxCommandlet(
 export async function forceAuthLogoutAll() {
   await commandlet.run();
 }
+
+export class AuthLogoutDefault extends LibraryCommandletExecutor<string> {
+  constructor() {
+    super(
+      nls.localize('force_auth_logout_default_text'),
+      'force_auth_logout_default',
+      OUTPUT_CHANNEL
+    );
+  }
+
+  public async run(
+    response: ContinueResponse<string>,
+    progress?: Progress<{
+      message?: string | undefined;
+      increment?: number | undefined;
+    }>,
+    token?: CancellationToken
+  ): Promise<boolean> {
+    try {
+      await removeUsername(response.data);
+    } catch (e) {
+      telemetryService.sendException(e.name, e.message);
+      return false;
+    }
+    return true;
+  }
+}
+
+export async function forceAuthLogoutDefault() {
+  const { username, isScratch, error } = await resolveDefaultUsername();
+  if (error) {
+    telemetryService.sendException(error.name, error.message);
+    notificationService.showErrorMessage('Logout failed to run');
+  } else if (username) {
+    const logoutCommandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      isScratch
```

Comment:
Can you please remind why we use a different gatherer for scratchOrgs?

Reply:
@jag-sfdc If you log out of a scratch org without taking prior steps to establish credentials then you will not be able to login again to the org.

Reply:
Aah, yes. I now remember from the demo discussion.

Reply:
Can you add this as a comment on why we are doing this.